src/sentry/static/sentry/app/utils/eventWaiter.tsx

```diff
@@ -61,10 +62,30 @@ class EventWaiter extends React.Component<Props, State> {
 
   pollHandler = async () => {
     const {api, organization, project, onIssueRecieved} = this.props;
-
-    const {firstEvent} = await api.requestPromise(
-      `/projects/${organization.slug}/${project.slug}/`
-    );
+    let firstEvent = null;
+
+    try {
+      ({firstEvent} = await api.requestPromise(
+        `/projects/${organization.slug}/${project.slug}/`
+      ));
+    } catch (resp) {
+      if (!resp) {
+        return;
+      }
+
+      // This means org or project does not exist, we need to stop polling
+      if (resp.status === 404) {
+        // TODO: Add some UX around this... redirect? error message?
```

Comment:
@EvanPurkhiser Can we do something here?

Reply:
Error message could be good. But it's probably not the responsibility of this component to do that. Maybe we can add a `onError` prop and components using this can wire up a toast notifiction

Reply:
Yeah I think let's do that.